### ISSUE TITLE
Zoom GUI is not shown

### ISSUE THREAD

**Issue by danbst** [NixOS/nixpkgs#43350]
## Issue description
Zoom GUI is not shown on current master (https://github.com/NixOS/nixpkgs/commit/1539167d215841a966b8395c1025d66812d63d31)
This started to happen after package refactoring (https://github.com/NixOS/nixpkgs/pull/42742, https://github.com/NixOS/nixpkgs/pull/42922, https://github.com/NixOS/nixpkgs/pull/43149), and not due to package upgrade (I've checked that).

I've documented some findings already https://github.com/NixOS/nixpkgs/pull/43149#issuecomment-403320716

ping @matthewbauer @tadfisher @shlevy @srhb @domenkozar and other Zoom users. Am I alone with this issue?
### Steps to reproduce
```
$ nix-env -if https://github.com/nixos/nixpkgs/archive/master.tar.gz -A zoom-us -p ~/zoom-us-nix-profile
$ ~/zoom-us-nix-profile/bin/zoom-us 
zoom started.
[CZPClientLogMgr::LogClientEnvironment] [MacAddr: 54:13:79:91:A6:8F][client: Linux][OS: NixOS 18.09.pre.5da8543 (Jellyfish)][Hardware: CPU Core:2 Frenquency:2.5 G Memory size:15951MB CPU Brand:Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz GPU Brand:][Req ID: ]
Linux Client Version is 2.2.128200.0702
QSG_RENDER_LOOP is 
XDG_CURRENT_DESKTOP = GNOME;   GDMSESSION = gnome3
Graphics Card Info:: 00:02.0 VGA compatible controller: Intel Corporation HD Graphics 620 (rev 02)
Zoom package arch is 64bit, runing OS arch is x86_64
AppIconMgr::systemDesktopName log Desktop Name: gnome3 
^C
```
At this moment no GUI (no greeting window) is shown.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.16.10, NixOS, 18.09.pre.5da8543 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1pre6148_a4aac7f`
 - channels(root): `"nixos-18.09pre142796.4b649a99d84, nixpkgs-18.09pre142651.95a8cb3ade1"`
 - channels(danbst): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by srhb** [comment]
I at least cannot reproduce this with `nix-env -iA nixpkgs.zoom-us` on nixpkgs 1539167

**Comment by tadfisher** [comment]
@danbst I installed to a profile using your reproduction steps, and it works fine here.

What is in your `LD_LIBRARY_PATH` in your shell? At least here it's:

```
$ echo $LD_LIBRARY_PATH 
/run/opengl-driver/lib:/run/opengl-driver-32/lib
```

**Comment by tadfisher** [comment]
I was able to reproduce this via:

```
nix run -f https://github.com/nixos/nixpkgs/archive/master.tar.gz -i -k HOME \
-k XDG_RUNTIME_DIR \
-k DISPLAY \
-k XAUTHORITY \
-k LD_LIBRARY_PATH \
zoom-us -c zoom-us
```

So there's some piece of the environment missing in the package.

**Comment by tadfisher** [comment]
This also reproduces the issue:
```
nix run -f https://github.com/nixos/nixpkgs/archive/master.tar.gz -u PATH zoom-us -c zoom-us
```
Currently investigating what `zoom-us` needs on the $PATH.

**Comment by coretemp** [comment]
Thanks for your efforts, but when writing a command can you (and everyone else) please be complete? I had time to quickly confirm what you wanted to say, but here I am telling you that it is not a valid test case.

```
error: Package ‘zoom-us-2.2.128200.0702’ in /nix/store/442xsjbqwld3dmddbjm0303xvlg4l3dg-master.tar.gz/pkgs/applications/networking/instant-messengers/zoom-us/default.nix:94 has an unfree license (‘unfree’), refusing to evaluate.
```

I wrote an example of a complete command in closed thread with zoom in the title started by myself, so it's not like I don't practice this myself.

**Comment by tadfisher** [comment]
@coretemp Sorry about that. As the package is unfree, you'll need `allowUnfree = true;` in `~/.config/nixpkgs/config.nix` or `NIXPKGS_ALLOW_UNFREE=1` in your environment. So a complete standalone test case would be:

```
NIXPKGS_ALLOW_UNFREE=1 nix run -f https://github.com/nixos/nixpkgs/archive/master.tar.gz -u PATH zoom-us -c zoom-us
```

**Comment by danbst** [comment]
@tadfisher does this reproduce issue too?

```
### bash
$ NIXPKGS_ALLOW_UNFREE=1 \
  nix-shell --pure -I nixpkgs=https://github.com/nixos/nixpkgs/archive/1539167.tar.gz \
            -p zoom-us \
            --run "
               export XDG_RUNTIME_DIR=$XDG_RUNTIME_DIR
               export LD_LIBRARY_PATH=/run/opengl-driver/lib
               zoom-us
            "
```
I've checked that commit 66e18c3 (last revision of old zoom-us expression) launches fine for me with this bash command.

**Comment by matthewbauer** [comment]
It sounds like it was caused by 81c2f270e535fbc2b7dc83c555b0d5625af004f4 using autoPatchelf? It could be that there is some incompatibility between what zoom-us is built with and what we have locally. Maybe we should just revert that part (but leave the version updates). I would much prefer to use Nixpkgs Qt here but there is a lot of unknown in how zoom-us is built. It might just be safest to revert to the old expression.

**Comment by tadfisher** [comment]
Turns out the issue was missing QML2 imports. This was discovered by comparing:

```
NIXPKGS_ALLOW_UNFREE=1 QT_DEBUG_PLUGINS=1 nix run -f  https://github.com/nixos/nixpkgs/archive/master.tar.gz \
-u PATH -c zoom-us
```

to

```
NIXPKGS_ALLOW_UNFREE=1 QT_DEBUG_PLUGINS=1 nix run -f  https://github.com/nixos/nixpkgs/archive/master.tar.gz \
-c zoom-us
```

and looking for libraries loaded by the latter but not the former.

One fix was to add a `QML2_IMPORT_PATHS` environment variable to the wrapper, but I found it simpler to re-use `libsForQt5.env` to build an environment with all the Qt dependencies symlinked in place and using its `qt.conf`. The package should be easier to maintain as a result since you don't have to know what feature each qt5 package is meant to provide.
